### PR TITLE
Consistent use of "private" vs "internal" terms for GitHub Apps

### PR DESCRIPTION
**Pull request by jamesmartin**
GitHub Apps can be either "public" or "private". We previously used the term "internal" and "private" interchangeably, leading to confusion. The UI has been updated to remove all references to "internal".

### Why:

**Closes https://github.com/github/docs-content/issues/4306**

### What's being changed:

This branch replaces all references to "internal" Apps with "private".

### Check off the following:
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (eg. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks so much for opening a PR @jamesmartin! I'll triage this for review ⚡

**Review by hotboyhot90 — COMMENTED** [review]
[](34YLgNjRjvKvbfW7aUGLuuEcjASHM4hQKR)

**Review by felicitymay — COMMENTED** [review]
Many thanks for raising the issue and also this PR. Your changes look great 💖 

My only question is when did you update the UI and which versions of GHES the new UI text is correct for. If GHES 2.21+ use the old terminology, then we'll need to add some versioning for these changes.

**Comment by jamesmartin** [comment]
@felicitymay thanks for the review! ✨ 

> My only question is when did you update the UI and which versions of GHES the new UI text is correct for. If GHES 2.21+ use the old terminology, then we'll need to add some versioning for these changes.

The UI was updated 10 days ago in https://github.com/github/github/pull/178328. It would not have been backported to older versions of GHES, so presumably we'll need to preserve the old terminology in those cases. How do we get started on that? 🤔

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Public installation flows have a landing page to enable other people besides the
 
 ### Private installation flow
 
-Private installation flows allow only the owner of a GitHub App to install it. Limited information about the GitHub App will still exist on a public page, but the **Install** button will only be available to organization administrators or the user account if the GitHub App is owned by an individual account. Private, or internal, GitHub Apps can only be installed on the user or organization account of the owner.
+Private installation flows allow only the owner of a GitHub App to install it. Limited information about the GitHub App will still exist on a public page, but the **Install** button will only be available to organization administrators or the user account if the GitHub App is owned by an individual account. Private GitHub Apps can only be installed on the user or organization account of the owner.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:26]:
```suggestion
Private installation flows allow only the owner of a GitHub App to install it. Limited information about the GitHub App will still exist on a public page, but the **Install** button will only be available to organization administrators or the user account if the GitHub App is owned by an individual account. Private{% if currentVersion ver_lt "enterprise-server@3.2" or currentVersion == "github-ae@latest" %}, or internal,{% endif %} GitHub Apps can only be installed on the user or organization account of the owner.
```

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
 3. Select the GitHub App whose installation option you want to change.
 ![App selection](/assets/images/github-apps/github_apps_select-app.png)
 {% data reusables.user-settings.github_apps_advanced %}
-5. Depending on the installation option of your GitHub App, click either **Make public** or **Make internal**.
+5. Depending on the installation option of your GitHub App, click either **Make public** or **Make private**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:38]:
```suggestion
5. Depending on the installation option of your GitHub App, click either **Make public** or **Make private{% if currentVersion ver_lt "enterprise-server@3.2" or currentVersion == "github-ae@latest" %}internal{% endif %}**.
```

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
...
 ![Button to change the installation option of your GitHub App](/assets/images/github-apps/github_apps_make_public.png)
-6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App internal**.
+6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App private**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:40]:
```suggestion
6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App private{% if currentVersion ver_lt "enterprise-server@3.2" or currentVersion == "github-ae@latest" %}internal{% endif %}**.
```

**Review by felicitymay — COMMENTED** [review]
Thanks for confirming. I've added what ought to be the right comments to ensure that this change is only visible for GitHub.com and GHES 3.2 onwards. The text for GHAE and GHES 3.1 and earlier versions should stay as it currently is, but we'll need to check that it behaves correctly on staging.

Will this change also be visible for the next update to GHAE? If so, we'll need some slightly more complex versioning.

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
 3. Select the GitHub App whose installation option you want to change.
 ![App selection](/assets/images/github-apps/github_apps_select-app.png)
 {% data reusables.user-settings.github_apps_advanced %}
-5. Depending on the installation option of your GitHub App, click either **Make public** or **Make internal**.
+5. Depending on the installation option of your GitHub App, click either **Make public** or **Make private{% if currentVersion ver_lt "enterprise-server@3.2" or currentVersion == "github-ae@latest" %}internal{% endif %}**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:38]:
```suggestion
5. Depending on the installation option of your GitHub App, click either **Make public** or **Make {% if currentVersion == "free-pro-team@latest" %}private{% else %}internal{% endif %}**.
```

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 5]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
...
 ![Button to change the installation option of your GitHub App](/assets/images/github-apps/github_apps_make_public.png)
-6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App internal**.
+6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App private{% if currentVersion ver_lt "enterprise-server@3.2" or currentVersion == "github-ae@latest" %}internal{% endif %}**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:40]:
```suggestion
6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App {% if currentVersion == "free-pro-team@latest" %}private{% else %}internal{% endif %}**.
```

**Review by felicitymay — COMMENTED** [review]
🤦🏻 Fixing those conditions

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 6]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
 3. Select the GitHub App whose installation option you want to change.
 ![App selection](/assets/images/github-apps/github_apps_select-app.png)
 {% data reusables.user-settings.github_apps_advanced %}
-5. Depending on the installation option of your GitHub App, click either **Make public** or **Make internal**.
+5. Depending on the installation option of your GitHub App, click either **Make public** or **Make {% if currentVersion == "free-pro-team@latest" %}private{% else %}internal{% endif %}**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:38]:
```suggestion
5. Depending on the installation option of your GitHub App, click either **Make public** or **Make {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.1" %}private{% else %}internal{% endif %}**.
```

File: content/developers/apps/making-a-github-app-public-or-private.md [hunk 7]
```diff
@@ -35,7 +35,7 @@ To change who can install the GitHub App:
...
 ![Button to change the installation option of your GitHub App](/assets/images/github-apps/github_apps_make_public.png)
-6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App internal**.
+6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App {% if currentVersion == "free-pro-team@latest" %}private{% else %}internal{% endif %}**.
```
Comment (felicitymay) [inline content/developers/apps/making-a-github-app-public-or-private.md:40]:
```suggestion
6. Depending on the installation option of your GitHub App, click either **Yes, make this GitHub App public** or **Yes, make this GitHub App {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.1" %}private{% else %}internal{% endif %}**.
```

**Review by felicitymay — APPROVED** [review]

**Comment by felicitymay** [comment]
I've had confirmation that this is correct for GHAE currently. So ready to merge.

**Pull request merged**